### PR TITLE
CIV-15073 CIV-15074 CIV-15075 CIV-15076 CIV-15077 CIV-15079 Sprint branch for CosC

### PR DESCRIPTION
**Pull request by Azam-Hmcts**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-15073
https://tools.hmcts.net/jira/browse/CIV-15074
https://tools.hmcts.net/jira/browse/CIV-15075
https://tools.hmcts.net/jira/browse/CIV-15076
https://tools.hmcts.net/jira/browse/CIV-15077
https://tools.hmcts.net/jira/browse/CIV-15079

### Change description ###
Sprint branch

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by LeonardoPalmeiro — APPROVED** [review]

File: src/main/modules/i18n/locales/cy.json [hunk 1]
```diff
@@ -2800,7 +2818,8 @@
         "SETTLING": "Court to make an order settling the claim by consent",
         "STRIKE_OUT": "Court to strike out all or part of the other parties' case without a trial",
         "SUMMARY_JUDGMENT": "Court to make a summary judgment on a case",
-        "VARY_JUDGMENT": "Vary a judgment"
+        "VARY_JUDGMENT": "Vary a judgment",
+        "CONFIRM_YOU_PAID_CCJ": "Confirm you've paid a judgment debt"
```
Comment (hmcts-version1-ignacio) [inline src/main/modules/i18n/locales/cy.json:2822]: This one is repeated. Same as "COMMON.ASK_FOR_PROOF_OF_DEBT_PAYMENT"

File: src/main/modules/i18n/locales/cy.json [hunk 2]
```diff
@@ -2723,6 +2725,22 @@
         },
         "UNAVAILABLE_DATES": "Are there any dates when you cannot attend a hearing within the next 3 months"
       },
+      "DEBT_PAYMENT": {
+        "DO_YOU_HAVE_EVIDENCE": "Do you have evidence of the debt payment?",
+        "YOU_WILL_NEED_EVIDENCE": "You’ll need to provide evidence that you’ve paid the debt.",
```
Comment (hmcts-version1-ignacio) [inline src/main/modules/i18n/locales/cy.json:2730]: There is an inconsistency between ’ and '

File: src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      const backLinkUrl = constructResponseUrlWithIdParams(req.params.id, GA_ASK_PROOF_OF_DEBT_PAYMENT_GUIDANCE_URL);
+      const defendantPaymentDate = new DefendantFinalPaymentDate(req.body.year, req.body.month, req.body.day);
+      const form: GenericForm<DefendantFinalPaymentDate> = new GenericForm<DefendantFinalPaymentDate>(defendantPaymentDate);
+
+      await form.validate();
+      if (form.hasErrors()) {
+        res.render(paymentDateViewPath, {form, title, cancelUrl, backLinkUrl });
+      } else {
+        try {
+
```
Comment (hmcts-version1-ignacio) [inline src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts:48]: Extra space

File: src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        next(error);
+      }
+    })
+  .post(
+    COSC_FINAL_PAYMENT_DATE_URL, async (req, res, next: NextFunction) => {
+      const cancelUrl = await getCancelUrl(req.params.id, null);
+      const backLinkUrl = constructResponseUrlWithIdParams(req.params.id, GA_ASK_PROOF_OF_DEBT_PAYMENT_GUIDANCE_URL);
+      const defendantPaymentDate = new DefendantFinalPaymentDate(req.body.year, req.body.month, req.body.day);
+      const form: GenericForm<DefendantFinalPaymentDate> = new GenericForm<DefendantFinalPaymentDate>(defendantPaymentDate);
+
```
Comment (hmcts-version1-ignacio) [inline src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts:42]: Extra line

File: src/test/a11y/ignored-urls.ts [hunk 5]
```diff
@@ -79,4 +79,6 @@ export const IGNORED_URLS = [
   urls.GA_PAY_ADDITIONAL_FEE_URL, //TODO: remove this once finished the page
   urls.GA_PROVIDE_MORE_INFORMATION_URL, //TODO: remove this once finished the page
   urls.TEST_SUPPORT_TOGGLE_FLAG_ENDPOINT,
+  urls.GA_CHECK_YOUR_ANSWERS_COSC_URL,  //TODO: remove this once finished the page
+  urls.GA_DEBT_PAYMENT_EVIDENCE_COSC_URL,
```
Comment (hmcts-version1-ignacio) [inline src/test/a11y/ignored-urls.ts:83]: GA_DEBT_PAYMENT_EVIDENCE_COSC_URL is developed, we should add the HTML

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Pull request merged**